Tests/SwiftPrettyPrintTests/Core/URLColoringWithDefaultColorThemeTests.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  URLColoringWithDefaultColorThemeTests.swift
+//  SwiftPrettyPrintTests
+//
+//  Created by ogasawara on 2021/02/08.
+//
+
+@testable import SwiftPrettyPrint
+import XCTest
+import SwiftParamTest
+import Curry
+
+/// Test the coloring of URL.
+/// - Note: The escape sequences used for the expected values of the test is as follows:
+///     - `\u{1B}`: escape characters starting all the escape sequences
+///     - `[0m`: Reset or normal. All attributes off
+///     - `[4m`: underline
+///     - `[33m`: make forground color yellow
+///     - `[34m`: make foreground color blue
+/// - SeeAlso:
+/// [ANSI escape code - Wikipedia](https://en.wikipedia.org/wiki/ANSI_escape_code)
+class URLColoringWithDefaultColorThemeTests: XCTestCase {
+
+    let describerWithDefaultColorTheme = PrettyDescriber(formatter: SinglelineFormatter(), theme: .default)
+    let exampleURL = "https://example.com"
+
+    override func setUpWithError() throws {}
+
+    override func tearDownWithError() throws {}
+
+    
+    /// URL coloring
+    /// - Throws: when the URL could not be generated from the string given in the `url` argument
+    func testURLColoring_non_debug() throws {
+        try _test(
+            describerWithDefaultColorTheme,
+            url: exampleURL,
+            expect: "\u{1B}[4m\u{1B}[34m\(exampleURL)\u{1B}[0m\u{1B}[4m\u{1B}[0m",
+            debug: false
+        )
+    }
+    
+    /// URL Coloring in detail
+    /// - Throws: when the URL could not be generated from the string given in the `url` argument
+    func testURLColoring_debug() throws {
+        try _test(
+            describerWithDefaultColorTheme,
+            url: exampleURL,
+            expect: "\u{1B}[33mURL\u{1B}[0m(\"\u{1B}[4m\u{1B}[34m\(exampleURL)\u{1B}[0m\u{1B}[4m\u{1B}[0m\")",
+            debug: true
+        )
+    }
+}
+extension URLColoringWithDefaultColorThemeTests {
+    /// Test PrettyDescriber
+    /// - Parameters:
+    ///   - describer: generates a string for output
+    ///   - url: a url string which is the target to generate strings
+    ///   - expect: the string that is expected to be generated
+    ///   - debug: whether generated string is a detailed version for debugging or not
+    /// - Throws: when the URL could not be generated from the string given in the `url` argument
+    private func _test(
+        _ describer: PrettyDescriber,
+        url string: String,
+        expect: String,
+        debug: Bool) throws {
+        let url = try XCTUnwrap(URL(string: string))
```

Comment:
In this case, I use `XCTUnwrap` function for assert that generated URL is not `nil`.

Reply:
This String of URL is given as pre-conditions of test at [L25](https://github.com/YusukeHosonuma/SwiftPrettyPrint/pull/155/files#diff-0bbb03358059b462570b3215cf996bb5b8f4165949fdaec4021290f9719a7e10R25).
So **force-unwrap** is always succeeded in tests.

Therefore I think to use **force-unwrap** is better than it.

```swift
let url = URL(string: string)!
```

or change type of variable `exempleURL` in L25 to `URL`.

```swift
let exampleURL = URL(string: "https://example.com")!
```

(I prefer the latter)

Reply:
This is reflected in 53b3077. Thanks!